### PR TITLE
:running: Move KCP code from 1 file into multiple logical ones

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:
After this PR is in, we'll need to work on moving the test code in multiple files and adding more tests, as well as documenting each function and what it does cc @wfernandes @sedefsavas 

/milestone v0.3.3
/kind cleanup
/area control-plane

### CODE REVIEW

File: controlplane/kubeadm/controllers/scale.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	controlplanev1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1alpha3"
+	"sigs.k8s.io/cluster-api/controlplane/kubeadm/internal"
+	"sigs.k8s.io/cluster-api/controlplane/kubeadm/internal/machinefilters"
+	capierrors "sigs.k8s.io/cluster-api/errors"
+	"sigs.k8s.io/cluster-api/util"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+func (r *KubeadmControlPlaneReconciler) initializeControlPlane(ctx context.Context, cluster *clusterv1.Cluster, kcp *controlplanev1.KubeadmControlPlane, controlPlane *internal.ControlPlane) (ctrl.Result, error) {
```
Comment (fabriziopandini) [inline controlplane/kubeadm/controllers/scale.go]: Should this be part of controller.go or helpers.go instead of scale.go?

Comment (vincepri) [reply controlplane/kubeadm/controllers/scale.go]: i put it here because it technically scales from 0 to 1 :)

Comment (fabriziopandini) [reply controlplane/kubeadm/controllers/scale.go]: ok. thans for the clarification

**Pull request merged**